src/components/ProgressIndicator/ProgressIndicator.jsx

```diff
@@ -1,141 +1,362 @@
-  <CarbonProgressIndicator {...others} />
-))`
-  &&& {
-    display: ${props => (!props.isVerticalMode ? `inline-flex` : '')};
+  const handleClick = () => {
+    onChange(id);
+  };
 
-    .${prefix}--progress-step--complete {
-      cursor: pointer;
+  const handleKeyDown = e => {
+    if (matches(e, [keys.Enter, keys.Space])) {
+      handleClick();
     }
-    .${prefix}--progress-step--incomplete {
-      cursor: pointer;
+  };
+
+  const getStepWidth = () => {
+    if (stepWidth != null && stepWidth > 0) {
+      return vertical
+        ? { height: `${stepWidth}rem`, minHeight: `${stepWidth}rem` }
+        : { width: `${stepWidth}rem`, minWidth: `${stepWidth}rem` };
     }
-  }
-`;
-
-const StyledProgressStep = styled(({ showLabel, stepWidth, isVerticalMode, ...others }) => (
-  <ProgressStep {...others} />
-))`
-  &&& {
-    width: ${props => (!props.isVerticalMode && props.stepWidth ? `${props.stepWidth}rem` : '')};
-    min-width: ${props =>
-      !props.isVerticalMode && props.stepWidth ? `${props.stepWidth}rem` : ''};
-    p {
-      display: ${props => (!props.showLabel ? 'none' : '')};
-
-      &:hover::after {
-        content: attr(data-label);
-        width: 100%;
-        position: absolute;
-        top: 100%;
+    return undefined;
+  };
+
+  const StepIcon = () => {
+    const completed = complete && !disabled && !invalid;
+    let value;
+
+    if (mainStep) {
+      if (completed) {
+        value = <CheckmarkOutline24 />;
+      } else if (invalid) {
+        value = <Warning24 />;
+      } else {
+        const number = (
+          <>
+            <text id="icon-numbered" x="50%" y="55%" dominantBaseline="middle" textAnchor="middle">
+              {stepNumber}
+            </text>
+          </>
+        );
+        if (current) {
+          value = <CircleFilled24>{number}</CircleFilled24>;
+        } else {
+          value = <RadioButton24>{number}</RadioButton24>;
+        }
       }
+    } else if (completed) {
+      value = <CheckmarkOutline16 />;
+    } else if (invalid) {
+      value = <Warning16 />;
+    } else if (current) {
+      value = <CircleFilled16 />;
+    } else {
+      value = <RadioButton16 />;
     }
-    ${props => {
-      const { isVerticalMode, stepWidth } = props;
-      return isVerticalMode
-        ? `
-        .${prefix}--progress-step-button {
-        flex-flow: initial;
-      }
-      height: ${stepWidth ? `${stepWidth}rem` : 'inherit'};
-      min-height: ${stepWidth ? `${stepWidth}rem` : '3.625rem'};
-      `
-        : '';
-    }}
-  }
-`;
-
-/**
- * This component extends the default Carbon ProgressIndicator.
- * It adds the ability to hideLabels on non-current steps and set a maximum stepWidth in pixels
- */
+
+    return (
+      <span className="icon" title={description}>
+        {value}
+      </span>
+    );
+  };
+
+  const StepLine = () => {
+    const classes = classnames({
+      [`line`]: !complete && !subStep,
```

Comment:
classes will all need a prefix and should follow the BEM syntax as outlined in Carbon repo